### PR TITLE
MessageListItem: use Hdy.Avatar

### LINKED ISSUE #510: Use Hdy.Avatar

### PR DESCRIPTION
**Pull request by danrabbit**
Fixes #510

### CODE REVIEW

File: src/MessageList/MessageListItem.vala [hunk 1]
```diff
@@ -68,10 +68,23 @@ public class Mail.MessageListItem : Gtk.ListBoxRow {
         loading_cancellable = new GLib.Cancellable ();
 
         style_context = get_style_context ();
-        style_context.add_class ("card");
+        style_context.add_class (Granite.STYLE_CLASS_CARD);
 
-        var avatar = new Granite.Widgets.Avatar.with_default_icon (48);
-        avatar.valign = Gtk.Align.START;
+        string parsed_address;
```
Comment (tintou) [inline src/MessageList/MessageListItem.vala:73]:
```suggestion
        string? parsed_address;
```

File: src/MessageList/MessageListItem.vala [hunk 2]
```diff
@@ -68,10 +68,23 @@ public class Mail.MessageListItem : Gtk.ListBoxRow {
...
+        string parsed_name;
```
Comment (tintou) [inline src/MessageList/MessageListItem.vala:74]:
```suggestion
        string? parsed_name;
```

File: src/MessageList/MessageListItem.vala [hunk 3]
```diff
@@ -68,10 +68,23 @@ public class Mail.MessageListItem : Gtk.ListBoxRow {
...
+
+        var camel_address = new Camel.InternetAddress ();
+        camel_address.unformat (message_info.from);
+        camel_address.ref ();
```
Comment (tintou) [inline src/MessageList/MessageListItem.vala:78]: You're leaking a reference here
```suggestion
```

**Review by tintou — CHANGES_REQUESTED** [review]

File: src/MessageList/MessageListItem.vala (hunk 3 above)
Comment (danrabbit) [reply src/MessageList/MessageListItem.vala:78]: Without the ref I get a crash from double free

**Review by tintou — APPROVED** [review]

**Pull request merged**